tests/SpecTests/transactions/mongos-pin-auto.json

```diff
@@ -3,7 +3,8 @@
     {
       "minServerVersion": "4.1.8",
       "topology": [
-        "sharded"
+        "sharded",
+        "load-balanced"
```

Comment:
There's one failure in the [task log](https://spruce.mongodb.com/task/mongo_php_library_test_loadBalanced__php_edge_versions~latest_stable_mongodb_versions~7.0_driver_versions~latest_stable_os~debian11_test_loadBalanced_patch_bb631434a42ac571eaab29918db3610c1bdc4044_6514b2b23627e08bcbc8dc72_23_09_27_22_55_04?execution=0&sortBy=STATUS&sortDir=ASC) to investigate on a load balancer (not serverless):

```
[2023/09/27 23:08:36.560] FAILURE:  (PHPUnit\Framework\ExpectationFailedException)
[2023/09/27 23:08:36.560] MongoDB\Tests\SpecTests\TransactionsSpecTest::testTransactions with data set "transactions/mongos-pin-auto: unpin after transient connection error on insertOne insert" (stdClass Object (...), array(stdClass Object (...)), array(stdClass Object (...), stdClass Object (...)), 'transaction-tests', 'test')
[2023/09/27 23:08:36.560] Failed asserting that null is not null.
[2023/09/27 23:08:36.560] /data/mci/2cd0f1f807b9a598faa6bcbc06dc1f76/src/tests/SpecTests/ErrorExpectation.php:145
[2023/09/27 23:08:36.560] /data/mci/2cd0f1f807b9a598faa6bcbc06dc1f76/src/tests/SpecTests/Operation.php:211
[2023/09/27 23:08:36.560] /data/mci/2cd0f1f807b9a598faa6bcbc06dc1f76/src/tests/SpecTests/TransactionsSpecTest.php:212
[2023/09/27 23:08:36.560] /data/mci/2cd0f1f807b9a598faa6bcbc06dc1f76/src/tests/SpecTests/TransactionsSpecTest.php:140
```

Reply:
Patch build running the test in isolation (passes): https://spruce.mongodb.com/version/65160090850e610a01d98d13/tasks

Patch build running the test in the full suite: https://spruce.mongodb.com/version/65160ba92fbabe5946c10a65/tasks

Reply:
I finally identified the root cause of this. The legacy spec test runner never asserted that a session was pinned when executing `targetedFailPoint`. Easy fix in [PHPLIB-1270](https://jira.mongodb.org/browse/PHPLIB-1270), which is now included in this PR.